settlement/dymension/dymension.go

```diff
@@ -751,3 +751,31 @@ func (c *Client) GetSignerBalance() (types.Balance, error) {
 
 	return balance, nil
 }
+
+func (c *Client) ValidateGenesisBridgeData(data rollapptypes.GenesisBridgeData) error {
+	var res *rollapptypes.QueryValidateGenesisBridgeResponse
+	req := &rollapptypes.QueryValidateGenesisBridgeRequest{
+		RollappId: c.rollappId,
+		Data:      data,
+	}
+
+	err := c.RunWithRetry(func() error {
+		var err error
+		res, err = c.cosmosClient.GetRollappClient().ValidateGenesisBridge(c.ctx, req)
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("rollapp client: validate genesis bridge: %w", err)
```

Comment:
in case it is not possible to validate genesis, for instance because hub rpc node is down, we need to add a clear message that the rollapp node needs to clear the data. even if the genesis is valid, but it is not possible to validate, this will return an error and the node will stop, but when starting again, the node will try to run again initchain but with it fail because initchain already happened, so it is necessary to clear the node to be able to run init chain again.

Reply:
good point
maybe we should query the hub for expected value, **before** executing `InitChain`

Reply:
but you need to run initchain to get the response from rdk module. i guess there are some workarounds to solve it (like retry forever till there is a response, and continue to that point in case of restart, etc) but by now i think is enough by being clear in an error message that, if it fails because not possible to query, it is necessary to clear the state (ie. remove data folder)

Reply:
> it will try to run again initchain but it will fail because initchain already happened

I tried it locally and did not see any errors. Anyway, I added a message that one needs to reset the node if there's a genesis validation error.

```go
return fmt.Errorf("Cannot validate genesis bridge data: %w. Please call `$EXECUTABLE dymint unsafe-reset-all` before the next launch to reset this node to genesis state.", err)
```

> maybe we should query the hub for expected value, before executing `InitChain`

I also thought about it. My main concern is that we move validation responsibility from the hub to dymint. Not sure if it's a problem though, but actual genesis validation happens in the hub on IBC transfer, and ideally we want to run the identical logic on InitChain. If we move it to dymint, it might not be identical for some reason. Plus we already almost have an RC in the hub, so don't want to disturb it with new PRs haha